src/net/ebpf/fd_ebpf_base.h

```diff
@@ -0,0 +1,41 @@
+#ifndef HEADER_fd_src_net_ebpf_fd_ebpf_base_h
+#define HEADER_fd_src_net_ebpf_fd_ebpf_base_h
+#if defined(__bpf__)
+
+/* eBPF base development environment
+
+   This file is a minimal port of fd_util_base.h to a non-hosted
+   Clang/LLVM eBPF little endian target.  This is a temporary measure
+   working around the fact that eBPF does not support libc yet, which is
+   a dependency of fd_util_base.  For documentations, see fd_util_base.h */
+
+/* Minimal libc-like environment **************************************/
+
+/* Not even the pre-processor definition is NULL is available on eBPF.
+   We define the bare minimum to resemble libc. */
+
+#define NULL (void *)0
```

Comment:
we need to be very careful here, since 0 is a valid location in packet contexts.
It's probably best if NULL is NOT defined, since it will always be misleading in those contexts

Reply:
test

Reply:
I'm not sure how that's relevant. I'm saying a test for, for example, p == (void*)0 isn't useful, because it may be true for a valid dereferenceable pointer.